### ISSUE TITLE
Rule Proposal: sfc-language

### ISSUE THREAD

**Issue by gewoonwoutje** [vuejs/eslint-plugin-vue#1458]
**Please describe what the rule should do:**

Limit the languages used in all Vue SFC files for template, script and style blocks.

**What category should the rule belong to?**

[ ] Enforces code style (layout)
[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ X] Other (please specify:)
Enforces code language.

**Provide 2-3 code examples that this rule should warn about:**
Using rule config:
 ```json5
{
    "script": "ts",
// "style": "scss",
// "template": "whatever",
}
```
```vue
// SomeComponent.vue

// Error: Missing attribute `lang="ts"`
<script>
// Error: Wrong lang attribute, expected `lang="ts"`
<script lang="js">

// Correct
<script lang="ts">
```

**Additional context**

We're using Typescript and Vue SFC files, but it is very easy to create a new component file and forget to add a `lang="ts"` attribute to the script tag, and missing out on some type checking. 

We also use `<style lang="scss">`.

**Comment by mauriciabad** [comment]
I find it very useful, I'd definately use it.

**Comment by gewoonwoutje** [comment]
I looked around a while ago to see if it was feasible for me to implement this. It seems this would require more information from the vue parser, as there is little info of the top level tag attributes  `<script> <template> <style>`.

@ota-meshi  is that a correct finding?
